### DISCUSSION TITLE
Struggling to translate a Groovy query to Gremlinq

### DISCUSSION THREAD

**Discussion by Dmo16** [Gremlinq/ExRam.Gremlinq#338] [Q&A] [answered]
Hi!

I'm having trouble translating this query: 

`g.V('id','11a54d3a-c375-45b5-8459-913695a0f').inE().hasLabel('OwnedByUser').outV().hasLabel('RaceTeam')`

Using the sample application as a guide, here's my method in .net:

`        public override async Task<List<GRaceTeam>> GetAllAsync(AimUser user)
        {
            var result = await Gremlin
                .V(user.Id)
                .InE<OwnedByUser>()
                .OfType<GRaceTeam>()
                .OutV<GRaceTeam>()
                .ToArrayAsync();
            return result.ToList();
        }`
        
 The groovy console code returns the correct records. The Gremlinq method returns null.
 
 What is the correct combination of fluent statements I should be trying?
 
 Thanks!

**Comment by yongkansage** [comment, marked as answer]
Try this
```
g.V<AimUser>(user.Id)
        .In<OwnedByUser>()
        .OfType<GRaceTeam>()
```

**Reply by Dmo16** [reply]
@yongkansage , that did help, but it turns out I had more than one issue!

I had named the class "GRaceTeam", G for graph, but the label on the Vertex was written out as "RaceTeam".

I updated my code to reflect this, and now I'm able to resolve correctly.

Thanks for the assist!
